File: Lezioni/11.Creazione-App-con-interfacce-gestite-da-tab/InterfacceGestiteDaTab/app/src/main/java/com/example/interfaccegestitedatab/MainActivity.java
package com.example.interfaccegestitedatab;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager2.widget.ViewPager2;

import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;

public class MainActivity extends AppCompatActivity {
    private TabLayout tabLayout;
    private ViewPager2 viewPager;
    private TabPagerAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tabLayout = findViewById(R.id.tab_layout);
        viewPager = findViewById(R.id.view_pager);

        // Crea l'adapter
        adapter = new TabPagerAdapter(this);
        viewPager.setAdapter(adapter);

        // Collega TabLayout a ViewPager2
        new TabLayoutMediator(tabLayout, viewPager, (tab, position) -> {
            switch (position) {
                case 0:
                    tab.setText("Home");
                    break;
                case 1:
                    tab.setText("Profilo");
                    break;
                case 2:
                    tab.setText("Impostazioni");
                    break;
            }
        }).attach();
    }

}

File: Lezioni/2.Panoramica-su-View-ViewGroup-widget-e-risorse-di-un-app/Esercizi/2.Localizzazione_e_Risorse/app/src/main/java/com/example/a2localizzazione_e_risorse/MainActivity.kt
package com.example.a2localizzazione_e_risorse

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Imposta il layout principale
        setContentView(R.layout.activity_main)

        // Trova il TextView e imposta il testo localizzato
        val welcomeTextView = findViewById<TextView>(R.id.welcome_text)
        welcomeTextView.text = getString(R.string.welcome_msg)
    }
}
